simple_use_case/aiida/suc_v2b1.py

```diff
@@ -0,0 +1,133 @@
+    filenames={"mesh": "mesh.msh"},
+    outputs=["*.xdmf", "*.h5"],
+)
+
+# ### solution of the poisson problem with fenics
+fenics_results, fenics_node = launch_shell_job(
+    "python",
+    arguments=[
+        "{script}",
+        "--mesh",
+        "{mesh}",
+        "--degree",
+        "2",
+        "--outputfile",
+        "poisson.pvd",
+    ],
+    files={
+        "script": "../source/poisson.py",
+        "mesh": meshio_results["mesh_xdmf"],
+        "mesh_h5": meshio_results["mesh_h5"],
+    },
+    filenames={"mesh": "mesh.xdmf", "mesh_h5": "mesh.h5"},
+    outputs=["poisson.pvd", "poisson000000.vtu"],
+)
+
+# ### postprocessing of the fenics job
+paraview_results, paraview_node = launch_shell_job(
+    "pvbatch",
+    arguments=["{script}", "{pvdfile}", "plotoverline.csv"],
+    files={
+        "script": "../source/postprocessing.py",
+        "pvdfile": fenics_results["poisson_pvd"],
+        "vtufile": fenics_results["poisson000000_vtu"],
+    },
+    filenames={"pvdfile": "poisson.pvd", "vtufile": "poisson000000.vtu"},
+    outputs=["plotoverline.csv"],
+)
+
+
+def read_domain_size():
+    stdout = gmsh_results["stdout"].get_content()
+    s = stdout.split("Used domain size:")[1]
+    size = float(s.split("Used mesh size")[0])
+    return str(size)
+
+
+def read_num_dofs():
+    stdout = fenics_results["stdout"].get_content()
+    ndofs = stdout.split("Number of dofs used:")[1]
+    return "".join(ndofs.split())
+
+
+# ### prepare latex macros
+macros, macros_node = launch_shell_job(
+    "python",
+    arguments=[
+        "{script}",
+        "--macro-template-file",
+        "{template}",
+        "--plot-data-path",
+        "{csvfile}",
+        "--domain-size",
+        read_domain_size(),
+        "--num-dofs",
+        read_num_dofs(),
+        "--output-macro-file",
+        "macros.tex",
+    ],
+    files={
+        "script": "../source/prepare_paper_macros.py",
+        "template": "../source/macros.tex.template",
+        "csvfile": paraview_results["plotoverline_csv"],
+    },
+    filenames={"csvfile": "plotoverline.csv"},
+    outputs=["macros.tex"],
+)
+
+# ### compile paper
+paper, paper_node = launch_shell_job(
+    "tectonic",
+    arguments=["{texfile}"],
+    files={
+        "texfile": "../source/paper.tex",
+        "macros": macros["macros_tex"],
+        "csvfile": paraview_results["plotoverline_csv"],
+    },
+    filenames={
+        "texfile": "paper.tex",
+        "macros": "macros.tex",
+        "csvfile": "plotoverline.csv",
+    },
+    outputs=["paper.pdf"],
+)
+
+# ### extract final PDF from database
+pdfWriter = PyPDF2.PdfFileWriter()
+output_file = open("./target.pdf", "wb")
+
+with paper["paper_pdf"].open(mode="rb") as handle:
+    output_file.write(handle.read())
+
+pdfWriter.write(output_file)
+
+breakpoint()
+# while at breakpoint, try viewing `target.pdf` --> not empty
+# after c(ontinue) and end of program `target.pdf` is empty ...
+output_file.close()
```

Comment:
Hi @dglaeser 
to clarify: the produced output PDF is the correct one and while the program is interrupted I can view "target.pdf" and it is not empty. However, after the program exits and I view "target.pdf" again it is empty ... (not sure what is happening)

Reply:
I fixed it. I should have just done that before posting, sorry.

Reply:
great, thanks a lot!!